crates/accelerate/src/synthesis/permutation/utils.rs

```diff
@@ -144,3 +144,18 @@ pub fn decompose_cycles(cycles: &Vec<Vec<usize>>) -> Vec<(usize, usize)> {
 
     swaps
 }
+
+//
+pub fn create_swap_layer(pattern: &ArrayView1<usize>, starting_point: usize) -> Vec<(usize, usize)> {
+    let mut pattern_vec = pattern.to_vec();
+    let num_qubits = pattern_vec.len();
+    let mut gates = Vec::new();
+
+    for j in (starting_point..num_qubits).step_by(2) {
+        if pattern_vec[j] > pattern_vec[(j + 1) % num_qubits] {
+            gates.push((j, (j + 1) % num_qubits));
+            pattern_vec.swap(j, (j + 1) % num_qubits);
+        }
+    }
```

Comment:
For this I believe that the higher bound should be `num_qubits-1` instead of `num_qubits` as per the original function. You told me you're doing the modulo `num_qubits` so it doesn't go out of bounds, but I think this should pass.
```suggestion
    for j in (starting_point..num_qubits - 1).step_by(2) {
        if pattern_vec[j] > pattern_vec[(j + 1)] {
            gates.push((j, (j + 1)));
            pattern_vec.swap(j, (j + 1));
        }
    }
```